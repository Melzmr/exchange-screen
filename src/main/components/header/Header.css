.header_container {
    display: flex;
    justify-content: space-between;
    background-color: green;
    position: fixed;
    top: 0;
    min-width: inherit;
    width: 100%;
}
