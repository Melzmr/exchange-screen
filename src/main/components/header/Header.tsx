import React from 'react';
import './Header.css';

export function Header() {
  return (
      <header className="jopa">
        <div>
          {'Cancel'}
        </div>
        <div>
          {'Pockets'}
        </div>
        <div>
          {'Exchange'}
        </div>
      </header>
  )
}
