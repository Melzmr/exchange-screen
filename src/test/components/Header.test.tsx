import React from 'react';
import {render} from '@testing-library/react';
import {Header} from '../../main/components/header/Header';
import {shallow} from 'enzyme';

describe('Header', () => {
  let comp: React.ReactElement;
  let fn: jest.Mock;

  beforeEach(() => {
    fn = jest.fn();
    comp = (
        <Header
            crossRate={'0.12'}
            selectedTop={{
              name: 'USD',
              value: '100',
              currency: '$',
            }}
            selectedBottom={{
              name: 'EUR',
              value: '50',
              currency: 'â‚¬',
            }}
            disabled={false}
            exchangeOnClick={fn}
        />
    );
  });

  it('renders Header component', () => {
    const wrapper = shallow(comp);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Header component with the same pockets', () => {
    const wrapper = shallow(
        <Header
            crossRate={'0.12'}
            selectedTop={{
              name: 'USD',
              value: '100',
              currency: '$',
            }}
            selectedBottom={{
              name: 'USD',
              value: '100',
              currency: '$',
            }}
            disabled={false}
            exchangeOnClick={fn}
        />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders exchange button in header', () => {
    const {getByText} = render(comp);
    const exchangeEl = getByText('Exchange');
    expect(exchangeEl).toBeInTheDocument();
  });

  it('renders cancel button in header', () => {
    const {getByText} = render(comp);
    const cancelEl = getByText('Cancel');
    expect(cancelEl).toBeInTheDocument();
  });

  it('renders crossRate in header', () => {
    const {getByText} = render(comp);
    const crossRate = getByText(/0.12/);
    expect(crossRate).toBeInTheDocument();
  });

  it(`it doesn't render cross rate if pockets are the same`, () => {
    const {queryByText} = render(
        <Header
            crossRate={'0.12'}
            selectedTop={{
              name: 'USD',
              value: '100',
              currency: '$',
            }}
            selectedBottom={{
              name: 'USD',
              value: '100',
              currency: '$',
            }}
            disabled={false}
            exchangeOnClick={fn}
        />,
    );
    const crossRate = queryByText(/0.12/);
    expect(crossRate).toBe(null);
  });

  it('calls exchangeOnClick', async () => {
    const wrapper = render(comp);
    const btn = await wrapper.findByText('Exchange');
    btn.click();
    expect(fn).toBeCalled();
  });

});


