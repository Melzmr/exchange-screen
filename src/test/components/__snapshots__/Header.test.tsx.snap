// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component 1`] = `
<div
  className="header_container"
>
  <header
    className="header_inner_container"
  >
    <div
      className="header_cancel"
    >
      Cancel
    </div>
    <div
      className="header_currency"
    >
      <span
        className="header_currency_symbol"
      >
        $
      </span>
      1 = 
      <span
        className="header_currency_symbol"
      >
        â‚¬
      </span>
      0.12
    </div>
    <div
      className="header_exchange"
    >
      <button
        className="header_exchange_button "
        disabled={false}
        onClick={[MockFunction]}
      >
        Exchange
      </button>
    </div>
  </header>
</div>
`;

exports[`Header renders Header component with the same pockets 1`] = `
<div
  className="header_container"
>
  <header
    className="header_inner_container"
  >
    <div
      className="header_cancel"
    >
      Cancel
    </div>
    <div
      className="header_exchange"
    >
      <button
        className="header_exchange_button "
        disabled={false}
        onClick={[MockFunction]}
      >
        Exchange
      </button>
    </div>
  </header>
</div>
`;
